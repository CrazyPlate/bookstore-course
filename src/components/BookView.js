import React from 'react';

export default class BookView extends React.Component {
   render() {
      return (
         <li className="bookView row" >
            <div className="col-md-2">
               <img src={this.props.book.image} width="75" height="100" alt={this.props.book.name}/>
            </div>
            <div className="col-md-6">
               <b>{this.props.book.name}</b><br />
               <i>{this.props.book.author}</i>
            </div>
            <div className="col-md-4">
               <button className="btn btn-secondary" onClick={ () => this.props.addToOrder( this.props.book ) }>Add to order</button>
            </div>
         </li>
      );
   }
}