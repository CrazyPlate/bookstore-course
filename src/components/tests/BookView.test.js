import React from 'react';
import BookView from '../BookView';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('BookView tests', () => {

   it('Snapshot matches', () => {
      const book = {
         name: "Book title"
      };
      const wrapper = shallow(<BookView book={book} />);
      expect(wrapper).toMatchSnapshot();
   })

   it('Title of the book and author are displayed as in props', () => {
      const book = {
         name: "Book title",
         author: "Author",
         image: "images/mort.jpg"
      };
      const wrapper = shallow(<BookView book={book} />);
      expect(wrapper.find('span').find('b').text()).toBe(book.name);
      expect(wrapper.find('span').find('i').text()).toBe(book.author);
   })
})