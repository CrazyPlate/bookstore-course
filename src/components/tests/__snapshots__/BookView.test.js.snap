// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookView tests Snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookView
    book={
      Object {
        "name": "Book title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Book title"
          height="100"
          width="75"
        />,
        <span>
          <b>
            Book title
          </b>
          <i />
        </span>,
        <button
          className="btn btn-secondary"
          onClick={[Function]}
        >
          Add to order
        </button>,
      ],
      "style": Object {
        "padding": "5px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Book title",
          "height": "100",
          "src": undefined,
          "width": "75",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <b>
              Book title
            </b>,
            <i />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Book title",
            },
            "ref": null,
            "rendered": "Book title",
            "type": "b",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add to order",
          "className": "btn btn-secondary",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add to order",
        "type": "button",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Book title"
            height="100"
            width="75"
          />,
          <span>
            <b>
              Book title
            </b>
            <i />
          </span>,
          <button
            className="btn btn-secondary"
            onClick={[Function]}
          >
            Add to order
          </button>,
        ],
        "style": Object {
          "padding": "5px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Book title",
            "height": "100",
            "src": undefined,
            "width": "75",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <b>
                Book title
              </b>,
              <i />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Book title",
              },
              "ref": null,
              "rendered": "Book title",
              "type": "b",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add to order",
            "className": "btn btn-secondary",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add to order",
          "type": "button",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
